.circleci/config.yml

```diff
@@ -32,6 +32,36 @@ workflows:
           tag: "latest,$CIRCLE_BRANCH,$CIRCLE_SHA1"
           # create the AWS ECR repo if it does not exist already.
           create-repo: true
+          lifecycle-policy-path: >
+            {
+              "rules": [
+                {
+                  "rulePriority": 1,
+                  "description": "Keep latest master image",
+                  "selection": {
+                    "tagStatus": "tagged",
+                    "tagPrefixList": ["master", "main"],
+                    "countType": "imageCountMoreThan",
+                    "countNumber": 5
+                  },
+                  "action": {
+                    "type": "expire"
+                  }
+                },
```

Comment:
IIUC, this is super dangerous; because we re-tag the `master` image on any CI run of `master`, the image that's actually being used in prod will get marked for expiration immediately after any build if that prod image is >90 days old. This is especially bad if for example, a PR gets merged, build, and pushed to the `master` tag, but flaky tests prevent it from being deployed. If it sits in that state, the in-use prod image will expire and we'll be unable to start new tasks.

Reply:
These are valid concerns -- after some discussion we've thought about mitigating these by tagging all images with $CIRCLE_BRANCH-$CIRCLE_BUILD_NUM, which should make the first rule protect any such image.

We've also increase the kept images to 8,000, i.e. 80% of the AWS ECR limit.